packages/flutter/lib/src/services/text_input.dart

```diff
@@ -908,6 +911,74 @@ abstract class TextInputClient {
   ///
   /// [TextInputClient] should cleanup its connection and finalize editing.
   void connectionClosed();
+
+  /// Requests that the client show the editing toolbar, for example when the
+  /// platform changes the selection through a non-flutter method such as
+  /// scribble.
+  void showToolbar();
```

Comment:
I worry about adding one more way to show the toolbar, since it's already pretty complicated figuring out what's causing it to show or not.  It does seem to be necessary in this case though...